libs/SalesforceSDKCore/SalesforceSDKCore/Classes/SFAppStyler.m

```diff
@@ -0,0 +1,210 @@
+ * Redistributions of source code must retain the above copyright notice, this list of conditions
+ and the following disclaimer.
+ * Redistributions in binary form must reproduce the above copyright notice, this list of
+ conditions and the following disclaimer in the documentation and/or other materials provided
+ with the distribution.
+ * Neither the name of salesforce.com, inc. nor the names of its contributors may be used to
+ endorse or promote products derived from this software without specific prior written
+ permission of salesforce.com, inc.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "SFAppStyler.h"
+
+@interface SFAppStyler ()
+
+@property (nonatomic, strong) NSMutableDictionary *fontMap;
+
+@end
+
+@implementation SFAppStyler
+
++(instancetype)sharedInstance {
+    static dispatch_once_t onceToken;
+    static SFAppStyler *styler = nil;
+    dispatch_once(&onceToken, ^{
+        styler = [[self alloc] init];
+    });
+    return styler;
+}
+
+- (instancetype)init {
+    self = [super init];
+    if (self) {
+        _fontMap = [[NSMutableDictionary alloc] init];
+        
+        // default values
+        _titleColor = [UIColor colorWithRed:0.024 green:0.110 blue:0.245 alpha:1.0];
+        _primaryAppColor = [UIColor colorWithRed:22.0/255.0 green:87.0/255.0 blue:205/255.0 alpha:1.0];
+        _actionableTextColor = [UIColor colorWithRed:0.333 green:0.525 blue:0.937 alpha:1.0];
+        _headerTextColor = [UIColor whiteColor];
+        _primaryTextColor = [UIColor blackColor];
+        _toolbarTextColor = nil;
+        _toolbarTextFont = nil;
+        [self updateDefaultSize];
+    }
+    return self;
+}
+
+- (void)updateDefaultSize {
+    self.headerFont =  [self fontWithSize:16.0 style:SFAppStylerFontStyleBold];
+    self.primaryTextFont = [self fontWithSize:14.0 style:SFAppStylerFontStyleRegular];
+    self.actionableFont = [self fontWithSize:14.0 style:SFAppStylerFontStyleRegular];
+}
+
+- (void)setFontName:(NSString * _Nullable)fontName forStyle:(SFAppStylerFontStyle)style {
+    if (fontName) {
+        self.fontMap[@(style)] = fontName;
+    } else {
+        [self.fontMap removeObjectForKey:@(style)];
+    }
+    [self updateDefaultSize];
```

Comment:
This is a bit odd to me.  Why do the different fonts get reset when you set a new font style-to-name configuration?  What if you've already customized the size of the fonts being reset?  You have to do it again?